### DISCUSSION TITLE
Run commands when the previous command has ran

### DISCUSSION THREAD

**Discussion by og900aero** [gokcehan/lf#1560] [Q&A] [answered]
I have a problem. The following code not working properly:

```
cmd select_filter &{{
	lf -remote "send push :glob-select<space>*"
	printf "$fs" >> ~/.local/share/lf/select_files
}}
```

The problem is that the printf command runs even if I hit enter because of the push command. So there should be a solution to jump to the printf command only when the push command has already been executed and I hit enter.

**Comment by joelim-work** [comment]
A better approach would be to have `select_filter` take arguments and forward them to `glob-select`:

```shell
cmd select_filter &{{
    lf -remote "send glob-select $1"
    lf -remote "send $id !printf ..."
}}
```

The only thing to watch out for is that commands inside `lf -remote` are run *asynchronously*, so any following commands in the script would be run immediately, which is probably not what you want.

**Comment by og900aero** [comment]
> A better approach would be to have `select_filter` take arguments and forward them to `glob-select`:
> 
> ```shell
> cmd select_filter &{{
>     lf -remote "send glob-select $1"
>     lf -remote "send $id !printf ..."
> }}
> ```
> 
> The only thing to watch out for is that commands inside `lf -remote` are run _asynchronously_, so any following commands in the script would be run immediately, which is probably not what you want.

But how do I give it to him? Because even though * is written out, this can always change. There are times when I transcribe to *.txt, or *.mp3, or *string* .
Then if I change the & sign to $, it doesn't wait for the end of the previous command either.

**Reply by joelim-work** [reply]
I'm not sure what you mean, but you would just type something like `:select_filter *.txt` on the command line, and the `*.txt` argument will be passed to `glob-select`.

**Comment by og900aero** [comment]
> ```shell
> lf -remote "send $id !printf ..."
> ```

I try the following code (used ! for debug, maybe error messages):

```
cmd select_filter !{{
    lf -remote "send glob-select $1"
    lf -remote "send $id !printf \"$fs\" >> ~/.local/share/lf/select_files"
}}
```
Then type in lf: select_filter *.txt, then select all txt, but doesnt print to select_files, and not show error messages.

**Reply by joelim-work** [reply, marked as answer]
You are very close, the `$` in `$fs` has to be escaped, otherwise it evaluates in the current shell.

**Comment by og900aero** [comment]
> You are very close, the `$` in `$fs` has to be escaped, otherwise it evaluates in the current shell.

:D Now, working great. Thx and Happy New Year!
